cmd/ck-server/ck-server.go

```diff
@@ -180,6 +180,18 @@ func dispatchConnection(conn net.Conn, sta *server.State) {
 			user.CloseSession(ci.SessionId, "Failed to connect to proxy server")
 			continue
 		}
+		err = localConn.(*net.TCPConn).SetKeepAlive(true)
```

Comment:
localConn could be a UDP conn. This timeout should be dealt with in Pipe

Reply:
This is TCP keepalive, not sending of data. I think in Pipe it is too late to set any socket options. I can just check if the connect is a TCPConn instead.